### PR TITLE
fix(startup): Q may fail if Toolkit fails

### PR DESCRIPTION
**Pull request by justinmk3**
Problem:
If Toolkit fails to start, then Q will also fail to start:

    2024-10-01 09:56:13.188 [error] aws.amazonq.refreshConnectionCallback: [
      [TypeError: Cannot read properties of undefined (reading 'getApi')
      at c:\…\dist\src\extensionNode.js:6093:284

Solution:
Check for undefined `exports`.

### CODE REVIEW

File: packages/core/src/test/shared/credentials/userCredentialsUtils.test.ts [hunk 1]
```diff
@@ -151,8 +142,8 @@ describe('UserCredentialsUtils', function () {
                 `creds.secretKey: "${profile.aws_access_key_id}" !== "${creds.secretKey}"`
             )
 
-            await assertDoesNotThrow(async () => await fs.checkPerms(credentialsFilename, 'r--'))
-            await assertDoesNotThrow(async () => await fs.checkPerms(credentialsFilename, '-w-'))
+            await assert.doesNotReject(async () => await fs.checkPerms(credentialsFilename, 'r--'))
```
Comment (jpinkney-aws) [inline packages/core/src/test/shared/credentials/userCredentialsUtils.test.ts:145]: Are these changes somehow related to this PR? Otherwise the change looks good to me

Comment (justinmk3) [reply packages/core/src/test/shared/credentials/userCredentialsUtils.test.ts:145]: no just a fixup for a recent commit
